### PR TITLE
Feature/513 research studies

### PR DESCRIPTION
**Pull request by twstuart**

### CODE REVIEW

File: root/api/src/main/java/org/patientview/api/controller/ResearchStudyController.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.inject.Inject;
+
+/**
+ * RESTful interface for the management and retrieval of News. NewsItems are made visible to specific Groups, Roles and
```
Comment (p-maks) [inline root/api/src/main/java/org/patientview/api/controller/ResearchStudyController.java]: You might need to refactor all the comments in this class, not related to news

File: root/api/src/main/java/org/patientview/api/model/NewsItem.java [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (p-maks) [inline root/api/src/main/java/org/patientview/api/model/NewsItem.java]: extra line remove please

File: root/api/src/main/java/org/patientview/api/service/ResearchService.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+public interface ResearchService {
+
+    /**
+     * Add a NewsItem.
+     *
+     * @param researchStudy Research Study item to add
+     * @return Long ID of the newly added NewsItem
+     * @throws ResourceNotFoundException
+     */
+    @RoleOnly(roles = {RoleName.SPECIALTY_ADMIN, RoleName.UNIT_ADMIN })
```
Comment (p-maks) [inline root/api/src/main/java/org/patientview/api/service/ResearchService.java]: extra space , RoleName.UNIT_ADMIN }) please remove

File: root/api/src/main/java/org/patientview/api/service/ResearchService.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    Long add(ResearchStudy researchStudy);
+
+    /**
+     * Delete a Research Study.
+     *
+     * @param researchItemId ID of ResearchStudy to delete
+     * @throws ResourceNotFoundException
+     * @throws ResourceForbiddenException
+     */
+    @RoleOnly(roles = {RoleName.SPECIALTY_ADMIN, RoleName.UNIT_ADMIN })
```
Comment (p-maks) [inline root/api/src/main/java/org/patientview/api/service/ResearchService.java]: extra space here as well

File: root/api/src/main/java/org/patientview/api/service/ResearchService.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    /**
+     * Get a single NewsItem.
+     *
+     * @param researchItemId ID of NewsItem to retrieve
+     * @return NewsItem object
+     * @throws ResourceNotFoundException
+     * @throws ResourceForbiddenException
+     */
+    @RoleOnly(roles = {RoleName.SPECIALTY_ADMIN, RoleName.UNIT_ADMIN })
```
Comment (p-maks) [inline root/api/src/main/java/org/patientview/api/service/ResearchService.java]: here extra space as well

File: root/api/src/main/java/org/patientview/api/service/impl/CodeServiceImpl.java [hunk 6]
```diff
@@ -627,4 +627,72 @@ public void saveCodeExternalStandard(CodeExternalStandard codeExternalStandard)
 
         return reduced;
     }
+
+
+    @Override
+    public List<BaseCode> searchAdminDiagnosisCodes(String searchTerm)
```
Comment (p-maks) [inline root/api/src/main/java/org/patientview/api/service/impl/CodeServiceImpl.java]: do we need to protect this if this for admins only?

File: root/api/src/main/java/org/patientview/api/service/impl/ResearchServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.patientview.api.util.ApiUtil.getCurrentUser;
+
+/**
```
Comment (p-maks) [inline root/api/src/main/java/org/patientview/api/service/impl/ResearchServiceImpl.java]: Fix comments, not related to news

File: root/api/src/test/java/org/patientview/api/service/ResearchServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
```
Comment (p-maks) [inline root/api/src/test/java/org/patientview/api/service/ResearchServiceTest.java]: comments in this class are not related to what it's says, also either remove created on and by, or update

File: root/webapp/package.json [hunk 9]
```diff
@@ -46,6 +46,7 @@
     "node": ">=0.10.0"
   },
   "scripts": {
-    "test": "grunt test"
+    "test": "grunt test",
+    "start": "bower install && grunt servelocal --port 8080"
```
Comment (p-maks) [inline root/webapp/package.json]: is this for local env only?

File: root/webapp/package.json [hunk 10]
```diff
@@ -46,6 +46,7 @@
...
   }
```
Comment (p-maks) [inline root/webapp/package.json]: Any new table added will require updating sql schema V1__Create_Schema.sql, under database

File: root/webapp/package.json (hunk 9 above)
Comment (twstuart) [reply root/webapp/package.json]: yeah, should mean people can just run the project without issues

File: root/api/src/main/java/org/patientview/api/service/impl/CodeServiceImpl.java (hunk 6 above)
Comment (twstuart) [reply root/api/src/main/java/org/patientview/api/service/impl/CodeServiceImpl.java]: Protected on interface level now
